OneSignalSDK/onesignal/in-app-messages/src/main/java/com/onesignal/inAppMessages/internal/backend/impl/InAppBackendService.kt

```diff
@@ -209,4 +198,75 @@ internal class InAppBackendService(
     ) {
         Logging.error("Encountered a $statusCode error while attempting in-app message $requestType request: $response")
     }
+
+    private suspend fun attemptFetchWithRetries(
+        baseUrl: String,
+        rywToken: String?,
+        sessionDurationProvider: () -> Long,
+    ): List<InAppMessage>? {
+        var attempts = 1
+        var retryLimit: Int? = null // Retry limit will be determined dynamically
+
+        while (retryLimit == null || attempts <= retryLimit + 1) {
```

Comment:
We can make this loop condition a bit simpler if we make this a `do { } while` loop instead. That will let us drop the `retryLimit == null ||` part and will also indicate to the reader we will do this logic at least once.

Reply:
Done.